### ISSUE TITLE
代码有问题

### ISSUE THREAD

**Issue by Daner-Wang** [VITA-Group/UVC#1]
依赖都已经安装，不存在依赖的问题，但是代码运行时还是各种乱七八糟的错，有些错误在编辑器里就能明显看到，希望上传能够正确运行的代码，以及作者所展示的最优结果的log文件。

**Comment by Daner-Wang** [comment]
![image](https://user-images.githubusercontent.com/21330065/160092863-3c84891b-f7bc-4fe8-9dc4-42410488808a.png)
We eliminated some errors in the code and tried to reproduce the author's experiment with the parameters given by the author, however,  the experimental results were far different from those in the paper. **In author‘s paper, the proportion of remained FLOPs in deit-tiny is 49.23%, while in our reproducing the proportion is more than 70%, which is confusing**. In addition, we also found that **the author's calculation of FLOPS might be wrong** in the given code.
![image](https://user-images.githubusercontent.com/21330065/160098378-d750ee96-883f-46ce-808f-48376374de94.png)

**Comment by billysx** [comment]
Sorry for the late reply. 
We promise we will finish cleaning the code by the end of the next week and upload the full fixed codes.

**Comment by billysx** [comment]
Thank you for pointing out the issues in our codebase, our response to your problems are as follows:
1. We’ve pushed a new version of our cleaned code without running errors after all dependencies are installed. Apologies for the running errors.
2. Thank you for pointing out that the calculation of FLOPs of the fourth term should be multiplied by r_ratio instead of r_ratio$^2$. Actually, the result of DeiT-Small and DeiT-Base are all using r_ratio instead of r_ratio$^2$ for calculation. We only preserve the experiment results of DeiT-Small, provided with a full-scheduled training log updated under the file "log/", please check. As seen in the screen shot, the best accuracy is 78.82% with remained FLOPs percentage 50.41% as the finetuned model.
3. As for the calculation of FLOPs, due to multi-version of organizing the code, the numbers for DeiT-Tiny are under investigation. We will provide an update as soon as possible.
<img width="656" alt="Screen Shot 2022-05-30 at 1 34 14 AM" src="https://user-images.githubusercontent.com/55985788/171046195-86f7dfa1-b125-43e8-ad6e-0a327cbdcc57.png">

**Comment by billysx** [comment]
Update Jul. 7th: new logs of DeiT-Tiny and DeiT-Base
For DeiT-Tiny, it’s 49.23% in the paper, the reproduce result is 48.45% at epoch 9 in the screenshot. Which is not exactly the same but are very close.

<img width="662" alt="Screen Shot 2022-07-05 at 3 57 48 PM" src="https://user-images.githubusercontent.com/55985788/177818102-12a354c8-de32-4372-9be5-3550df002d15.png">
<img width="506" alt="Screen Shot 2022-07-05 at 3 59 16 PM" src="https://user-images.githubusercontent.com/55985788/177818108-2423ad67-fdeb-468f-8ea5-97d6dd8e85c7.png">

For DeiT-Base, it’s 45.5% in the paper, the reproduce result has a 42% model at epoch 17, and a 50% model at epoch 19.
Although these two results are not exactly the same numbers as it is in the paper(randomness), these two proves that the problem of cannot compressing to 50% FLOPs is not there.
For DeiT-Base, using a different z-learning-rate schedule "5,10,15,20,25" can get the result.

<img width="650" alt="Screen Shot 2022-07-05 at 3 59 27 PM" src="https://user-images.githubusercontent.com/55985788/177818109-55a0bdd4-b4cc-43ba-a049-18c38f27ce25.png">
<img width="1076" alt="Screen Shot 2022-07-05 at 4 01 21 PM" src="https://user-images.githubusercontent.com/55985788/177818111-817fae7f-ff6a-476c-a8b0-35f0679ef349.png">
<img width="534" alt="Screen Shot 2022-07-05 at 4 02 24 PM" src="https://user-images.githubusercontent.com/55985788/177818112-a1d78769-25fb-49fe-8868-cd5c6b5f1f59.png">

The logs that are mentioned are under file log/.
